docs/pattern-matching/as.md

```diff
@@ -59,9 +73,56 @@ actor Main
     end
 ```
 
+You can do the same with interfaces as well. In the example below, we have an Array of `Any` which is an interface where we want to try wash any entries that conform to the `Critter` interface.
+
+```pony
+actor Main
+  new create(env: Env) =>
+    let anys = Array[Any ref].>push(Wombat).>push(Capybara)
+    for any in anys.values() do
+      try
+        env.out.print((any as Critter).wash())
+      end
+    end
+```
+
+Note, All the `as` examples above could be written using a `match` statement where a failure to match results in `error`. For example, our last example written to use `match` would be:
+
+```pony
+actor Main
+  new create(env: Env) =>
+    let anys = Array[Any ref].>push(Wombat).>push(Capybara)
+    for any in anys.values() do
+      try
+        match any
+        | let critter: Critter =>
+          env.out.print(critter.wash())
+        else
+          error
+        end
+      end
+    end
+```
+
+Thinking of the `as` keyword as "an attempt to match that will error if not matched" is a good mental mode to have. If you don't care about handling the "not matched" case that causes an error when using `as`, you can rewrite an `as` to use match without an error like:
```

Comment:
I don't know of "mental mode" or "mental model" was the intended term, but I think the latter is more natural here.

```suggestion
Thinking of the `as` keyword as "an attempt to match that will error if not matched" is a good mental model to have. If you don't care about handling the "not matched" case that causes an error when using `as`, you can rewrite an `as` to use match without an error like:
```